packages/core/src/domain/configuration.ts

```diff
@@ -200,33 +222,58 @@ export function buildCookieOptions(userConfiguration: UserConfiguration) {
   return cookieOptions
 }
 
-function getEndpoint(type: string, conf: TransportConfiguration, source?: string) {
+function getEndpoint(type: EndpointType, conf: TransportConfiguration, source?: string, withLegacyHost = false) {
   const tags =
     `sdk_version:${conf.sdkVersion}` +
     `${conf.env ? `,env:${conf.env}` : ''}` +
     `${conf.service ? `,service:${conf.service}` : ''}` +
     `${conf.version ? `,version:${conf.version}` : ''}`
-  const datadogHost = `${type}-http-intake.logs.${conf.site}`
+  const datadogHost = getHost(type, conf, withLegacyHost)
   const host = conf.proxyHost ? conf.proxyHost : datadogHost
   const proxyParameter = conf.proxyHost ? `ddhost=${datadogHost}&` : ''
   const applicationIdParameter = conf.applicationId ? `_dd.application_id=${conf.applicationId}&` : ''
   const parameters = `${applicationIdParameter}${proxyParameter}ddsource=${source || 'browser'}&ddtags=${tags}`
 
   return `https://${host}/v1/input/${conf.clientToken}?${parameters}`
 }
+function getHost(type: EndpointType, conf: TransportConfiguration, useLegacyDomain: boolean) {
+  if (!useLegacyDomain) {
+    const subdomains = {
+      [EndpointType.BROWSER]: 'logs',
+      [EndpointType.RUM]: 'rum',
+      [EndpointType.TRACE]: 'trace',
+    }
+    if (conf.site === INTAKE_SITE[Datacenter.US]) {
+      return `${subdomains[type]}.browser-intake-datadoghq.com`
+    }
+    if (conf.site === STAGING_INTAKE_SITE[Datacenter.US]) {
+      return `${subdomains[type]}.browser-intake-datad0g.com`
+    }
```

Comment:
since we will need to enable other host when available, a check against something like `NEW_INTAKE_DOMAIN_ALLOWED_SITES = [...]` could be easier to maintain and could clarify the intent